instrumentation/kafka-clients/src/main/java/brave/kafka/clients/KafkaTracing.java

```diff
@@ -132,9 +135,12 @@ public KafkaTracing build() {
     this.processorExtractor = propagation.extractor(KafkaPropagation.GETTER);
     this.producerInjector = propagation.injector(KafkaProducerRequest::setHeader);
     this.consumerInjector = propagation.injector(KafkaConsumerRequest::setHeader);
+    this.propagationKeys = new LinkedHashSet<>(propagation.keys());
+    // When Extra Fields or similar are in use, the result != TraceContextOrSamplingFlags.EMPTY
+    this.emptyExtraction = propagation.<Map<String, String>>extractor(Map::get)
```

Comment:
this was a bug.. I searched for all cases we (like me) did incorrect comparisons against EMPTY

Reply:
Does this sort of bug only affect Kafka? If the pattern is generic enough, one idea is something like `Propagation.isEmpty(TraceContextOrSamplingFlags)` so we don't need to repeat this logic as much.